src/org/sosy_lab/common/collect/OrderStatisticMapTestSuite.java

```diff
@@ -0,0 +1,369 @@
+          .add(Maps.immutableEntry("pp", "Vhh"))
+          .add(Maps.immutableEntry("zz", "Vaa"))
+          .build();
+
+  public abstract static class OrderStatisticMapFactory extends TestStringSortedMapGenerator {
+
+    @Override
+    protected abstract OrderStatisticMap<String, String> create(Entry<String, String>[] pEntries);
+
+    protected abstract OrderStatisticMap<String, String> create(
+        List<Entry<String, String>> pEntries);
+  }
+
+  private OrderStatisticMapFactory factory;
+
+  public OrderStatisticMapTestSuite(OrderStatisticMapFactory pFactory) {
+    factory = pFactory;
+  }
+
+  @SuppressWarnings("unchecked")
+  private OrderStatisticMap<String, String> createMap() {
+    return factory.create(Collections.emptyList());
+  }
+
+  @SuppressWarnings("unchecked")
+  private OrderStatisticMap<String, String> createMap(List<Entry<String, String>> pEntries) {
+    return factory.create(pEntries);
+  }
+
+  private static <K, V> boolean containsEntry(Map<K, V> pMap, Entry<K, V> pEntry) {
+    return pMap.containsKey(pEntry.getKey()) && pEntry.getValue().equals(pMap.get(pEntry.getKey()));
+  }
+
+  private static <K, V> void putEntry(Map<K, V> pMap, Entry<K, V> pEntry) {
+    pMap.put(pEntry.getKey(), pEntry.getValue());
+  }
+
+  private static <K, V> void removeEntry(Map<K, V> pMap, Entry<K, V> pEntry) {
+    Assert.assertEquals(pMap.get(pEntry.getKey()), pMap.remove(pEntry.getKey()));
+  }
+
+  @Test
+  public void testEquals() {
+    OrderStatisticMap<String, String> l1 = createMap();
+    @Var OrderStatisticMap<String, String> l2 = createMap();
+
+    // Check that the map sorts its elements
+    Assert.assertEquals(l1, l2);
+    for (int i = ELEMS.size() - 1; i >= 0; i--) {
+      l2.put(ELEMS.get(i).getKey(), ELEMS.get(i).getValue());
+    }
+    for (int i = 0; i < ELEMS.size(); i++) {
+      l1.put(ELEMS.get(i).getKey(), ELEMS.get(i).getValue());
+    }
+    Assert.assertEquals(l1, l2);
+
+    // Check the map property
+    l2 = createMap();
+    for (int i = ELEMS.size() - 1; i >= 0; i--) {
+      l2.put(ELEMS.get(i).getKey(), ELEMS.get(i).getValue());
+    }
+    for (int i = 0; i < ELEMS.size(); i++) {
+      l2.put(ELEMS.get(i).getKey(), ELEMS.get(i).getValue());
+    }
+    Assert.assertEquals(l1, l2);
+  }
+
+  @Test
+  public void testSerialize() {
+    OrderStatisticMap<String, String> l = createMap();
+    SerializableTester.reserializeAndAssert(l);
+
+    for (int i = 100000; i >= 0; i--) {
+      l.put(String.valueOf(i), String.valueOf(i));
+    }
+    SerializableTester.reserializeAndAssert(l);
+  }
+
+  @Test
+  public void testSubmapView_mutation() {
+    NavigableMap<String, String> map = createMap(ELEMS);
+    NavigableMap<String, String> subMap =
+        map.subMap(ELEMS.get(1).getKey(), true, ELEMS.get(2).getKey(), true);
+
+    Entry<String, String> toAdd = ELEMS_BELOW.get(2);
+
+    putEntry(subMap, toAdd);
+    Assert.assertTrue(containsEntry(subMap, toAdd));
```

Comment:
`assertTrue` and `assertFalse` create hard-to-debug test cases and should be avoided. The various Truth methods are much better.

For consistency, it might make sense to switch to Truth for all assertions in these classes.

Furthermore, these methods can be statically imported.